source/dub/dub.d

```diff
@@ -738,8 +738,10 @@ class Dub {
 				name/version.
 			force_remove = Forces removal of the package, even if untracked
 				files are found in its folder.
+			resolveVersion = Callback to select package version.
 	*/
-	void remove(string package_id, string version_, PlacementLocation location, bool force_remove)
+	void remove(string package_id, string version_, PlacementLocation location, bool force_remove,
+				scope size_t delegate(in Package[] packages) resolveVersion=null)
```

Comment:
What about adding a second overload instead, which removes the `version_` argument? The original one could then be implemented in terms of the resolver callback. Another change that might be worth it would be to change the callback to `bool delegate(in Package)`, so that it can opt to remove multiple packages at once, too.

Nitpick: "resolve_version" to match the naming convention of other parameters.